src/modules/jsx.ts

```diff
@@ -86,38 +93,45 @@ export function wrapWithComponent(fullPath, jsx): ProcessedSelection {
     componentMembers: new Set(),
   };
 
+  // TODO Requires Review
   const visitor = {
     Identifier(astPath) {
       let isMember = !!astPath.findParent(() => (
         (astPath.node.type === 'MemberExpression') ||
         (astPath.parent.type === 'ObjectProperty' && astPath.parent.value.type !== 'Identifier') ||
         astPath.isArrowFunctionExpression(astPath.node)
       ));
-      if (!isMember) {
+      if (!isMember && !astPath.node.wasVisited) {
         componentProperties.argumentProps.add(astPath.node.name);
       }
     },
-    MemberExpression(path) {
-      if (!path.node.wasVisited && t.isThisExpression(path.node.object)) {
-        if (path.parent.property && (path.node.property.name === 'props' || path.node.property.name === 'state')) {
+    MemberExpression(astPath) {
+      if (!astPath.node.wasVisited && t.isThisExpression(astPath.node.object)) {
+        if (astPath.parent.property && (astPath.node.property.name === 'props' || astPath.node.property.name === 'state')) {
           // props or state = path.node.property.name;
-          if (path.node.property.name === 'props') {
-            componentProperties.memberProps.add(path.parent.property.name);
+          if (astPath.node.property.name === 'props') {
+            componentProperties.memberProps.add(astPath.parent.property.name);
           } else {
-            path.node.property.name = 'props';
-            componentProperties.state.add(path.parent.property.name);
+            astPath.node.property.name = 'props';
+            componentProperties.state.add(astPath.parent.property.name);
           }
+        } else if (t.isThisExpression(astPath.node.object)) {
```

Comment:
@williamluke4 can you explain what case does this part handle?

Reply:
![glean](https://user-images.githubusercontent.com/10773719/49855768-3572bd80-fdff-11e8-9cac-7f6bec9e5a07.gif)

Reply:
@borislit Let me know if that cleared it up

Reply:
Oh Amazing!!! Its bug fix.  Can you create a separate PR? I want to keep sem-ver and release it ASAP. Thanks! @williamluke4